src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceProgress.cs

```diff
@@ -60,19 +70,60 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
                 _pendingProgress = new PendingProgress();
             }
 
-            _pendingProgress.Update(sourceId, record);
-
             if (_progPane == null)
             {
-                // This is the first time we've received a progress record. Create a progress pane, then show it.
+                // This is the first time we've received a progress record
+                // Create a progress pane
+                // Set up a update flag
+                // Create a timer for updating the flag
 
                 _progPane = new ProgressPane(this);
+
+                if (_progPaneUpdateTimer == null && _progPane != null)
```

Comment:
However, `_progPane` could be null when execution reaches here, because another thread might be calling `ResetProgress` on the same `ConsoleHostUserInterface` instance (e.g. another runspace is created using the same host).
This is a race condition caused by moving the lock within this method.